homeassistant/components/tile/__init__.py

```diff
@@ -33,6 +33,31 @@ async def async_setup_entry(hass, entry):
     hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id] = {}
     hass.data[DOMAIN][DATA_TILE][entry.entry_id] = {}
 
+    # The existence of shared Tiles across multiple accounts requires an entity ID
+    # change:
+    #
+    # Old: tile_{uuid}
+    # New: {username}_{uuid}
+    #
+    # Find any entities with the old format and update them:
+    ent_reg = entity_registry.async_get(hass)
+    for entity in [
+        e
+        for e in ent_reg.entities.values()
+        if e.platform == DOMAIN
+        and not e.unique_id.startswith(entry.data[CONF_USERNAME])
```

Comment:
So if I have 2 entries... the second entry will never match the first in terms of user name and gets converted?

Reply:
@frenck Not sure I totally understand your question. I did see an issue (I shouldn't be checking the domain; I should be checking the entry ID), so check out my latest and if you still have a question, let me know!

Reply:
Yeah that change is exactly fixing the issue I tried to explain 👍